src/main/java/com/xiaomi/infra/pegasus/client/PegasusTableInterface.java

```diff
@@ -1129,4 +1171,56 @@ public PegasusScannerInterface getScanner(
    */
   public List<PegasusScannerInterface> getUnorderedScanners(
       int maxScannerCount, ScanOptions options) throws PException;
+
+  /**
+   * send batch request no-atomically, but terminate immediately if any error occurs. Generally, it
+   * is for no need response result operations such as {@link
+   * com.xiaomi.infra.pegasus.client.request.BatchSet}, usage example see {@link
+   * com.xiaomi.infra.pegasus.example.BatchSample}
+   *
+   * @param batch batch request, detail see {@link Batch}, default implement contains {@link
+   *     com.xiaomi.infra.pegasus.client.request.BatchSet}, {@link
+   *     com.xiaomi.infra.pegasus.client.request.BatchGet} and so on
+   * @param timeout timeout, throw exception if exceed the value
+   * @param <Req> generic for request
+   * @param <Res> generic for response
+   * @throws PException any error occurs will throw exception
+   */
+  public <Req, Res> void batch(Batch<Req, Res> batch, int timeout /*timeout*/) throws PException;
+
+  /**
+   * send batch request no-atomically, but terminate immediately if any error occurs. Generally, it
+   * is for need response result operations such as {@link
+   * com.xiaomi.infra.pegasus.client.request.BatchGet}, usage example see {@link
+   * com.xiaomi.infra.pegasus.example.BatchSample}
+   *
+   * @param batch batch request, detail see {@link Batch}, default implement contains {@link
+   *     com.xiaomi.infra.pegasus.client.request.BatchSet}, {@link
+   *     com.xiaomi.infra.pegasus.client.request.BatchGet} and so on
+   * @param responses the batch response, mustn't be null
+   * @param timeout timeout, throw exception if exceed the value
+   * @param <Req> generic for request
+   * @param <Res> generic for response
+   * @throws PException any error occurs will throw exception
+   */
+  public <Req, Res> void batch(Batch<Req, Res> batch, List<Res> responses, int timeout /*timeout*/)
+      throws PException;
+
+  /**
+   * send batch request no-atomically, try wait for all requests done until timeout even if some
+   * other error occurs. usage example see {@link com.xiaomi.infra.pegasus.example.BatchSample}
+   *
+   * @param batch batch request, detail see {@link Batch}, default implement contains {@link
+   *     com.xiaomi.infra.pegasus.client.request.BatchSet}, {@link
+   *     com.xiaomi.infra.pegasus.client.request.BatchGet} and so on
+   * @param responses mustn't be null, if the single request success, the response is pair(null,
+   *     result), otherwise is pair(PException, null)
+   * @param timeout timeout, throw exception if exceed the value
+   * @param <Req> generic for request
+   * @param <Res> generic for response
+   * @throws PException throw exception if timeout
+   */
+  public <Req, Res> void batchWaitAllComplete(
+      Batch<Req, Res> batch, List<Pair<PException, Res>> responses, int timeout /*timeout*/)
+      throws PException;
```

Comment:
Stop adding new interfaces. You are killing your effort to decouple batching from basic API. 

```
Batch gets = new Batch(table, requests, timeout);
gets.commitAllComplete();
```

```
Batch gets = new Batch(requests);
table.batchWaitAllComplete(gets);
```

There's no difference here. Use the former if it's the way you encourage.